android/java/org/chromium/chrome/browser/crypto_wallet/fragments/ApproveTxBottomSheetDialogFragment.java

```diff
@@ -189,6 +186,7 @@ public void onDismiss(DialogInterface dialog) {
         super.onDismiss(dialog);
         if (mApprovedTxObserver != null) {
             if (mRejected || mApproved) {
+                // TODO(pav): 28/07/22 rename to callback, it's not an observer
```

Comment:
it's not a callback as well, it's more like delegate

Reply:
why?

Reply:
callback is something that you pass somewhere to be called. Delegate is something that is triggered on some action for delegating some kind of work.

Reply:
so this is being passed here https://github.com/brave/brave-core/blob/b7bd7a4376ede1c7737665da24cf568dde13eb9a/android/java/org/chromium/chrome/browser/crypto_wallet/fragments/ApproveTxBottomSheetDialogFragment.java#L114 from a parent/container so i guess callback is fine though have updated the comment for both to make a wise call later